src/main/java/csci331/team/red/dao/CharacterDAO.java

```
@@ -0,0 +1,462 @@
+		String DOB;
+
+		int year = (int) ((Math.random() * 100) % 80) + 1917;
+		int month = (int) ((Math.random() * 12) % 12) + 1;
+
+		int day;
+		// list of months with 31 days
+		int[] longMonth = new int[] { 1, 3, 5, 7, 8, 10, 12 };
+		// list of months with 30 days
+		int[] shortMonth = new int[] { 4, 6, 9, 11 };
+
+		if (Arrays.asList(longMonth).contains(month)) {
+			day = (int) (Math.random() * ((31 - 1) + 1));
+
+		} else if (Arrays.asList(shortMonth).contains(month)) {
+			day = (int) (Math.random() * ((30 - 1) + 1));
+
+			// February
+		} else {
+			day = (int) (Math.random() * ((28 - 1) + 1));
+		}
+
+		String monthString;
+		switch (month) {
+		case 1:
+			monthString = "Jan";
+			break;
+		case 2:
+			monthString = "Feb";
+			break;
+		case 3:
+			monthString = "Mar";
+			break;
+		case 4:
+			monthString = "Apr";
+			break;
+		case 5:
+			monthString = "May";
+			break;
+		case 6:
+			monthString = "Jun";
+			break;
+		case 7:
+			monthString = "Jul";
+			break;
+		case 8:
+			monthString = "Aug";
+			break;
+		case 9:
+			monthString = "Sep";
+			break;
+		case 10:
+			monthString = "Oct";
+			break;
+		case 11:
+			monthString = "Nov";
+			break;
+		case 12:
+			monthString = "Dec";
+			break;
+		default:
+			monthString = "Invalid month";
+			break;
+		}
+
+		DOB = year + "-" + monthString + "-" + day;
+
+		return DOB;
+	}
+
+	/**
+	 * Randomly generate a Drivers Licence number
+	 * @return DriversID
+	 * @author melany
+	 */
+	public static String getDriversID() {
+		return String
+				.valueOf(((int) (Math.random() * ((9999999 - 1000000) + 1))));
+	}
+
+	/**
+	 * Fetches a random id based on the database entity
+	 * @param tableName
+	 * @return random id
+	 * @author melany
+	 */
+	public static int randomID(String tableName) {
+		PreparedStatement statement = null;
+		try {
+			statement = conn.connection.prepareStatement("SELECT ID FROM "
+					+ tableName + " ORDER BY Random() LIMIT 1");
+		} catch (SQLException e) {
```

This exception is swallowed, but will continue to break the game. Perhaps just re-throw the exception as a `RuntimeException`.

If you think that an SQL error can be handled, or recovered from, then perhaps throw some kind of non-eql exception...

The same goes for all of the other SQLExceptions.